xdrgen/src/spec/xdr.rustpeg

```diff
@@ -136,14 +136,17 @@ union_default -> Decl
 const_def -> Defn
     = const id:identifier eq v:constant semi    { Defn::Const(id, v) }
 
-type_def -> (String, Type)
-    = typedef d:declaration semi               { if let Decl::Named(name, ty) = d { (name, ty) } else { return Failed } }
-    / enum id:identifier e:enum_body semi      { (id, Type::Enum(e)) }
-    / struct id:identifier s:struct_body semi  { (id, Type::Struct(s)) }
-    / union id:identifier u:union_body semi    { (id, Type::union(u)) }
+type_def -> Defn
+    = typedef d:declaration semi               { if let Decl::Named(name, ty) = d {
+                                                    if ty.is_syn() { Defn::Typesyn(name, ty) }
+                                                    else { Defn::Typespec(name, ty) }
```

Comment:
the typesyn and typespec terms i just made up. the distinction is to cover the difference between type synonyms and what my brain was thinking of as type specifications. looking at RFC 4506, this is maybe type definitions vs. type declarations (i feel like that is C usage as well, or perhaps i'm confused). i wanted to stay away from typedefs though, because there is a keyword called that and now i'm confused which way is which, come to think of it.

anyway, that's all philosophy/design. the actual bug that i was way too laconic about was that xdr-codec has this trait impl:
```rust
impl<Out: Write, T: Pack<Out>> Pack<Out> for Vec<T> {
    #[inline]
    fn pack(&self, out: &mut Out) -> Result<usize> {
        (&self[..]).pack(out)
    }
}
```
and xdr specs such as the following:
```
```